sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -126,14 +127,56 @@ class SessionCatalog(
   }
 
   /**
-   * A cache of qualified table names to table relation plans.
-   */
+    * A cache of qualified table names to table relation plans.
+    * Accessing tableRelationCache directly is not recommended,
+    * since it will introduce exposures to guava libraries.
+    */
   val tableRelationCache: Cache[QualifiedTableName, LogicalPlan] = {
```

Comment:
This seems to have been introduced in 2.2, can you make it private instead, and change all call sites?